packages/framework/ai-collab/src/test/explicit-strategy/typeGeneration.spec.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+// eslint-disable-next-line import/no-internal-modules
+import { createIdCompressor } from "@fluidframework/id-compressor/internal";
+// eslint-disable-next-line import/no-internal-modules
+import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils/internal";
+// eslint-disable-next-line import/order
+import {
+	SchemaFactory,
+	SharedTree,
+	TreeViewConfiguration,
+	// eslint-disable-next-line import/no-internal-modules
+} from "@fluidframework/tree/internal";
+
+// eslint-disable-next-line import/no-internal-modules
+import { doesNodeContainArraySchema } from "../../explicit-strategy/typeGeneration.js";
+
+const factory = SharedTree.getFactory();
+const sf = new SchemaFactory("test");
+
+class Todo extends sf.object("Todo", {
+	title: sf.string,
+	completed: sf.boolean,
+}) {}
+
+class TestTodoAppSchema extends sf.object("TestTodoAppSchema", {
+	title: sf.string,
+	description: sf.string,
+	todos: sf.array(Todo),
+}) {}
+
+const initialAppState = {
+	title: "My First Todo List",
+	description: "This is a list of todos",
+	todos: [
+		{
+			title: "Task 1",
+			completed: true,
+		},
+		{
+			title: "Task 2",
+			completed: true,
+		},
+	],
+};
+
+describe("Type Generation", () => {
```

Comment:
Related to a comment further up, I'm curious about what would happen in a test if you pass an object and none of its properties are arrays, but one of them is an object where one of _its_ properties is an array.